src/compiler/rust.rs

```diff
@@ -215,12 +219,15 @@ fn parse_dep_info<T>(dep_info: &str, cwd: T) -> Vec<String>
 }
 
 /// Run `rustc --print file-names` to get the outputs of compilation.
-fn get_compiler_outputs<T>(creator: &T, executable: &str, arguments: &[String], cwd: &str) -> SFuture<Vec<String>>
+fn get_compiler_outputs<T>(creator: &T, executable: &str, arguments: &[String], cwd: &str,
+                           env_vars: &Vec<(String, String)>) -> SFuture<Vec<String>>
     where T: CommandCreatorSync,
 {
     let mut cmd = creator.clone().new_command_sync(executable);
     cmd.args(&arguments)
         .args(&["--print", "file-names"])
+        .env_clear()
+        .envs(env_vars.iter().map(|&(ref k, ref v)| (k, v)))
```

Comment:
Yeah, that's a sensible idea. I actually went back and changed this to add `envs` on `RunCommand` to make it slightly less verbose, but encapsulating all this commonality seems like a good idea.